src/rnnlm/sampling-lm-estimate.cc

```diff
@@ -539,7 +539,7 @@ BaseFloat SamplingLmEstimator::GetProbForWord(
   // compute the probability from lowest to highest order.
   KALDI_ASSERT(word > 0 && word < static_cast<int32>(unigram_probs_.size()));
   BaseFloat ans = unigram_probs_[word];
-  for (size_t i = 0; i < states.size(); i++) {
+  for (int32 i = states.size() - 1; i >= 0; i--) {
```

Comment:
nit: While you're at it, our coding style requires that loops only use pre-{increment, decrement} form,  {`++i`, `--i`} resp. BTW, this is another idiomatic C[++] reverse array iteration, up to you:
```suggestion
  for (int32 i = states.size(); --i >= 0;) {
```